chromium_src/third_party/blink/renderer/core/loader/mixed_content_checker.cc

```diff
@@ -0,0 +1,46 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "third_party/blink/renderer/core/loader/mixed_content_checker.h"
+
+#define BRAVE_MIXED_CONTENT_CHECKER_IS_MIXED_CONTENT              \
+  if (auto result = IsMixedContentForOnion(security_origin, url)) \
+    return *result;
+
+#include "src/third_party/blink/renderer/core/loader/mixed_content_checker.cc"
+
+#undef BRAVE_MIXED_CONTENT_CHECKER_IS_MIXED_CONTENT
+
+namespace blink {
+
+namespace {
+
+template <typename T>
+bool IsOnion(const T& obj) {
+  constexpr const char kOnion[] = ".onion";
+  return obj.Host().EndsWith(kOnion) && (obj.Protocol() == url::kHttpsScheme ||
+                                         obj.Protocol() == url::kHttpScheme);
```

Comment:
Should we be including the websockets scheme here? (I'm not sure if `ws://address.onion` is feasible.)

Reply:
might as well handle websockets scheme